python/common/org/python/types/Float.java

```diff
@@ -352,39 +365,16 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
                 if (other_val == 0) {
                     throw new org.python.exceptions.ZeroDivisionError("integer division or modulo by zero");
                 } else {
-                    double result = this.value % other_val;
-                    if (other_val > 0 && result < 0) {
-                        // second operand is positive, ensure that result is positive
-                        result += other_val;
-                    } else if (other_val < 0 && result > 0) {
-                        // second operand is negative, ensure that result is negative
-                        result += other_val; // subtract other_val, which is negative
-                    }
-                    if (other_val < 0 && result >= 0
-                            && Double.doubleToRawLongBits(result) != NEGATIVE_ZERO_RAW_BITS) {
-                        result *= -1;
-                    }
+                    // Reference: http://stackoverflow.com/a/4412200
```

Comment:
Given that Stack Overflow content might change over time, yes - include the paragraph as an explanation.

The code itself looks fine; however, it needs tests. There's a very specific set of combinations of values that need to be tested, and the only way to ensure that they are *all* reliably tested is an explicit test for this behavior.